### PR TITLE
feat: Additional tasks for batchReplayer to enable reactive pipelines

### PR DESCRIPTION
**Pull request by rubenhgaws**
The purpose of this new feature is to transform the batchReplayer Step Function into a _reactive_ pipeline.

Once all the batchReplayer's tasks finish, the Step Function's output state contains enough information to trigger **additional** steps / tasks that can be easily chained re-using AWS Resources already deployed.

### CODE REVIEW

File: core/test/unit/data-generator/batch-replayer.test.ts [hunk 1]
```diff
@@ -138,4 +138,17 @@ test("BatchReplayer should create a step function", () => {
   template.resourceCountIs("AWS::StepFunctions::StateMachine", 1);
 });
 
+test("BatchReplayer should not have additionalStepFunctionTasks by default", ()=> {
+  expect(batchReplayer.additionalStepFunctionTasks).toBeUndefined();
+});
+
+test("BatchReplayer should allow additionalStepFunctionTasks as empty list", ()=> {
```
Comment (vgkowski) [inline core/test/unit/data-generator/batch-replayer.test.ts:145]: can we add a test to check the tasks are added to the CFN template?

**Review by vgkowski — COMMENTED** [review]
e2e test should also be modified with additional tasks

File: core/src/data-generator/batch-replayer.ts [hunk 2]
```diff
@@ -142,6 +171,13 @@ export class BatchReplayer extends Construct {
    */
   public readonly vpc?: IVpc;
 
+  /**
```
[the remark this answers is not in the document]
Comment (rubenhgaws) [reply core/src/data-generator/batch-replayer.ts:174]: Excellent point. It doesn't need to be public.

File: core/test/unit/data-generator/batch-replayer.test.ts (hunk 1 above)
Comment (rubenhgaws) [reply core/test/unit/data-generator/batch-replayer.test.ts:145]: Yes.

File: core/src/data-generator/batch-replayer.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (vgkowski) [reply core/src/data-generator/batch-replayer.ts:174]: By default members are public in Typescript

[the remark this answers is not in the document]
Comment (rubenhgaws) [reply core/src/data-generator/batch-replayer.ts:174]: Oh. I tried it private but was unable to test the property value, that is why I removed the public - to no effect!
Suggestions on how to test private properties?

[the remark this answers is not in the document]
Comment (vgkowski) [reply core/src/data-generator/batch-replayer.ts:174]: For unit testing, we generally test the CFN template from the stack. Is it possible to check the JSON description of the AWS::StepFunctions::StateMachine resource contains the tasks you added?

[the remark this answers is not in the document]
Comment (lmouhib) [reply core/src/data-generator/batch-replayer.ts:174]: Here is an example (not for step function) that shows how to pick a CFN resource and assert against it.
 https://github.com/aws-samples/aws-analytics-reference-architecture/blob/d4e5fb16bae226907c847e85d2d6e8a26830403a/core/test/unit/notebook-platform/notebook-platform.test.ts#L69

**Pull request merged**